### ISSUE TITLE
Failure to load dependency lib for azure ML 

### ISSUE THREAD

**Issue by nehalahmad2** [Azure/MachineLearningNotebooks#1759]
File "/structure/azureml-app/score.py", line 23, in run
inputs_dc.collect(data) #this call is saving our input data into Azure Blob
File "/azureml-envs/sklearn-1.0/lib/python3.8/site-packages/azureml/monitoring/modeldatacollector.py", line 457, in collect
if (not self._cloud_enabled or self._handle == -1) and not self._debug:
AttributeError: 'ModelDataCollector' object has no attribute '_cloud_enabled'

**Comment by JoelCanteroGCO** [comment]
I have the same error.

**Comment by SonHoangTR** [comment]
Same error with me. Not sure if there is any workaround ? Thanks

**Comment by JoelCanteroGCO** [comment]
> Same error with me. Not sure if there is any workaround ? Thanks

Are you using ModelDataCollector class?

**Comment by SonHoangTR** [comment]
> > Same error with me. Not sure if there is any workaround ? Thanks
> 
> Are you using ModelDataCollector class?

Yes, and I got exactly same error. This might be a problem with 3.8 only ?

**Comment by JoelCanteroGCO** [comment]
> > > Same error with me. Not sure if there is any workaround ? Thanks
> > 
> > 
> > Are you using ModelDataCollector class?
> 
> Yes, and I got exactly same error. This might be a problem with 3.8 only ?

I have opened a support ticket with Microsoft and it seems that the error is relationed with >1.40 SDK version. I have tried with 1.38 and it works well.

**Comment by jnesfield** [comment]
> 

Can you provide specfics on how you got this to work such as python version and all that jazz?

**Comment by RohitRanga12** [comment]
**> I have opened a support ticket with Microsoft and it seems that the error is relationed with >1.40 SDK version. I have tried with 1.38 and it works well.**

How did you change the sdk version?

**Comment by JoelCanteroGCO** [comment]
Just installing another version of azureml-core.

Like this: pip install azureml-core==1.40

**Comment by shmahe** [comment]
Please use the latest `azureml-core` and the latest base image. Using these latest versions will eliminate this issue.

**Comment by dsiu1** [comment]
> Please use the latest `azureml-core` and the latest base image. Using these latest versions will eliminate this issue.

Which versions are you referring to? When I use azureml-core==1.50.0 and the base image 'mcr.microsoft.com/azureml/openmi4.1.0-ubuntu20.04:20230509.v1' I still get the _cloud_enabled error. My current requirements do not allow me to downgrade to ubuntu 18.04. Any insight would be appreciated, thanks!
